src/main/scala/sigmastate/eval/CostingRules.scala

```diff
@@ -627,4 +627,23 @@ trait CostingRules extends SigmaLibrary { IR: RuntimeCosting =>
 
   object SigmaDslBuilderCoster extends CostingHandler[SigmaDslBuilder]((obj, m, costedArgs, args) => new SigmaDslBuilderCoster(obj, m, costedArgs, args))
 
+  /** Costing rules for SNumericType methods */
+  class SNumericTypeCoster(obj: RCosted[AnyVal], method: SMethod, costedArgs: Seq[RCosted[_]], args: Seq[Sym]) extends Coster[AnyVal](obj, method, costedArgs, args) {
```

Comment:
Looks like it is better to use separate Coster classes and objects for each numeric type SByte, Short, SInt,